doc/SAI-Proposal-ACL-1.md

```diff
@@ -0,0 +1,89 @@
+SAI access control lists (ACL) enhancements for SAI 1.0 release
+-------------------------------------------------------------------------------
+
+ Title       | SAI ACL Model - Enhancements
+-------------|----------------------
+ Authors     | Cavium Inc.
+ Status      | In review
+ Type        | Standards track
+ Created     | 08/09/2016
+ SAI-Version | 1.0.0
+
+-------------------------------------------------------------------------------
+
+## Overview ##
+SAI Access Control Lists (ACL) object implements ACL management functions. In SAI 0.9.1 through 0.9.5 versions, SAI ACL contained three types of objects, ACL table, ACL entry and ACL counter. The ACL table contains a number of ACL entries. Each ACL table defines a set of unique matching fields for all its ACL entries. A packet can match rules in different ACL tables and take non-conflicting actions from all the matched rules. However, within an ACL table, if a packet matches multiple rules, only the actions from the rule of highest priority are executed. ACL counters can also be created and attached to an ACL entry in order to counter the number of packets or bytes that match the ACL entry. The initial version of ACL table object has several ambiguities and limitations. We propose the following enhancements to address some of those and propose a generic and simple ACL model for operators:
```

Comment:
@itaibaz @lguohan @AshokDaparthi @atitjain : currently here is what I have listed in the updated specification and documentation. Lets discuss these details before the review so that we have clear agenda and asks if there needs to be a change or provision put in. But the changes for not addressing non-conflicting behavior are very specifically coming from the previous ACL review we had in AUG 2016, and listing them here: 

**

> ### ACL Table and ACL Group Match Behavior                                      
> Within one ACL GROUP TABLE only one ACL entry will be hit which is based on the table priority as well as ACL entry priority within the table. Since only on ACL Group (or ACL Table) can be associated with a bind point, it is clear that only one entry (if hit) has corresponding actions to be taken. SAI 1.0.0 does not support resolution of non-conflicting actions across various ACL Groups. However, considering the behavioral pipeline (model) for SAI 1.0.0, multiple bind points can be valid for a specific flow. There are use cases when multiple ACL Tables and/or ACL Groups are assigned at different bind points, but the very first bind point with a valid ACL entry (if hit - irrespective of an ACL Table or ACL Group) takes precedence over rest of the entries. This avoids resolving non-conflicting actions across various ACL Groups.

**